### PR TITLE
Package reboot

### LINKED ISSUE #11: Support for environment variables
Instead of having a `config.yml` in the working directory, how would you feel about two environment variables with this information?

https://github.com/rstudio/rscloud/blob/85b286847dbe30db19b09a4dde5dd38b1aa4b6f3/R/setup.R#L12-L14

### LINKED ISSUE #12: Retrieve authorization state via whoami
It would be great to see the login credentials active in the _R_ session's `.global` environment, c.f. 

https://github.com/rstudio/rscloud/blob/85b286847dbe30db19b09a4dde5dd38b1aa4b6f3/R/setup.R#L3

Something like:

```
rscloud_whoami()
# User: user@example.edu
# CLIENT_ID: 981....AFG
# CLIENT_SECRET: 11Z...GGA
# Hourly API Calls: 1/100
```

### LINKED ISSUE #22: Set up tests

### LINKED ISSUE #23: Consider making token request implicit
Basically ask for it on the first API call that requires auth instead of calling `initialize_token()`

### PR DESCRIPTION
**Pull request by kevinykuo**
This PR is a reboot of the package with major API changes. Usage examples can be found in the `README`. The main design decisions are as follows:

- In naming functions, we use `noun_verb()` throughout, where `noun` (always singular) tells the type of the first argument and not necessarily the REST endpoint. The space listing function has been renamed to `rscloud_space_list()`.
- Since the *space* concept is the entry point into most of the functionality, it has been implemented as an R6 class (primarily to maintain a `status` attribute).
- Authentication now uses environment variables (`RSCLOUD_CLIENT_ID` and `RSCLOUD_CLIENT_SECRET`) instead of a `config.yml` in the project directory. Secrets are no longer cached in package globals, and spends a minimal amount of time in process memory. (Closes #11)
- Authentication is now implicit, i.e. users don't need to invoke a specific function to get authenticated. (Closes #23)
- For testing, the API URL and host can be set via the `rscloud_{api_url/host}_set()` functions or by setting the environment variables `RSCLOUD_API_URL` and `RSCLOUD_HOST`. We can re-introduce a config file in the future for specifying e.g. instance sizes.
- The features suppported in this change include
    - Spaces
        - List
    - Space members
        - List
        - Add
        - Remove
    - Space roles
        - List
    - Invitations
        - List
        - Resend
        - Rescind
    - Projects
        - List
    - User
        - whoami (Closes #12)
- Tests have been added (Closes #22)

### CODE REVIEW

File: R/auth.R [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  invisible(NULL)
+}
+
+#' Who Am I?
+#'
+#' Displays the current API user, authenticating if needed.
+#'
+#' @export
+rscloud_whoami <- function() {
+  response <- rscloud_rest("/users/me")
```
Comment (kevinykuo) [inline R/auth.R]: @ssinnott are we cool w/ exposing this? This is the only new endpoint we're hitting for this change.

**Comment by kevinykuo** [comment]
@coatless @mine-cetinkaya-rundel would be great if y'all can give this a whirl sometime!

File: R/auth.R (hunk 1 above)
Comment (ssinnott) [reply R/auth.R]: I think that looks like a sensible add! Thanks!

**Comment by coatless** [comment]
Quick glance over the restructured code, I think the `invitation_send()` functionality will be confused w.r.t. `space_member_add()`. Perhaps the earlier function could be renamed? 

I'll dig a bit deeper tomorrow.

**Comment by kevinykuo** [comment]
I think this is pretty well marinated, merging!

**Pull request merged**